data-service/src/main/java/uk/gov/gchq/palisade/service/data/web/DataController.java

```diff
@@ -72,37 +66,35 @@ public DataController(final DataService dataService,
      * @param dataRequest the request to read the data from a leafResource
      * @return a stream of bytes representing the contents of the resource
      */
-    @PostMapping(value = "/read/chunked", consumes = "application/json", produces = "application/octet-stream")
-    public ResponseEntity<StreamingResponseBody> readChunked(@RequestBody final DataRequest dataRequest) {
+    @PostMapping(value = "/read/chunked", consumes = "application/json", produces = "application/json")
+    public ResponseEntity<StreamingResponseBody> readChunked(
+            @RequestBody final DataRequest dataRequest) {
         LOGGER.info("Invoking read (chunked): {}", dataRequest);
-        DataReaderRequest readerRequest = dataService.authoriseRequest(dataRequest)
-                .thenApply(maybeReadRequest -> maybeReadRequest
-                        .orElseThrow(() -> new ForbiddenException(
-                                String.format("The token '%s' is not authorised to access the leafResource '%s'", dataRequest.getToken(), dataRequest.getLeafResourceId()))))
-                .join();
 
-        StreamingResponseBody stream = (OutputStream outputStream) -> {
-            Pair<AtomicLong, AtomicLong> recordsAudit = dataService.read(readerRequest, outputStream);
-            AuditSuccessMessage successMessage = auditService.createSuccessMessage(dataRequest, readerRequest, recordsAudit.getFirst().get(), recordsAudit.getSecond().get());
-            auditService.auditSuccess(dataRequest.getToken(), successMessage);
-        };
+        HttpStatus httpStatus = HttpStatus.ACCEPTED;
+        StreamingResponseBody stream = null;
```

Comment:
- [ ] Could possibly refactor this and the further down stuff into:

```
if (firstErrorMessage != null) {
  // audit
  return new ResponseEntity(null, HttpStatus.INTERNAL_SERVER_ERROR)
} else {
  // set stream
  return new ResponseEntity((StreamingResponseBody) stream, HttpStatus.ACCEPTED)
}
```
if  that reads better

Reply:
I like the idea, there are a couple of things that could be done better.  The reluctance is in how to refactor the "//set stream" section.  We have the possibility of a success and an error message.  For that I would want to step back and spend sometime on getting it right.  It does at least work.